Google.Api.Gax.Grpc/Rest/RestChannel.cs

```diff
@@ -186,16 +186,36 @@ internal Metadata GetHeaders()
 
             internal Status GetStatus()
             {
-                var grpcStatus = OriginalResponseMessage.StatusCode switch
+                var httpStatus = (int) OriginalResponseMessage.StatusCode;
+                var grpcStatus = (httpStatus / 100) switch
                 {
-                    HttpStatusCode.OK => StatusCode.OK,
-                    HttpStatusCode.Unauthorized => StatusCode.Unauthenticated,
-                    HttpStatusCode.Forbidden => StatusCode.PermissionDenied,
-                    HttpStatusCode.BadRequest => StatusCode.InvalidArgument,
-                    HttpStatusCode.InternalServerError => StatusCode.Internal,
-                    HttpStatusCode.NotFound => StatusCode.NotFound,
-                    HttpStatusCode.Conflict => StatusCode.FailedPrecondition,
-                    // TODO: Others!
+                    // All 2xx responses map to OK
+                    2 => StatusCode.OK,
```

Comment:
+1 to this.

The Apiary libraries marks all 2xx as success, if we didn't do this here, we could get the exact same operation succeed in Apiary but "fail" in REGAPIC.